src/nativeMain/kotlin/commands/PayInvoice.kt

```diff
@@ -0,0 +1,73 @@
+package commands
+
+import IResultWriter
+import api.IEclairClientBuilder
+import kotlinx.cli.ArgType
+import kotlinx.coroutines.runBlocking
+import types.PayInvoiceResult
+import types.Serialization
+
+class PayInvoiceCommand(
+    private val resultWriter: IResultWriter,
+    private val eclairClientBuilder: IEclairClientBuilder
+) : BaseCommand(
+    "payinvoice",
+    "Pays a BOLT11 invoice. In case of failure, the payment will be retried up to maxAttempts times."
+) {
+    private val invoice by option(
+        ArgType.String,
+        description = "The invoice you want to pay"
+    )
+    private val amountMsat by option(
+        ArgType.Int,
+        description = "Amount to pay if the invoice does not have one"
+    )
+    private val maxAttempts by option(
+        ArgType.Int,
+        description = "Max number of retries"
+    )
+    private val maxFeeFlatSat by option(
+        ArgType.Int,
+        description = "Fee threshold to be paid along the payment route"
+    )
+    private val maxFeePct by option(
+        ArgType.Int,
+        description = "Max percentage to be paid in fees along the payment route (ignored if below maxFeeFlatSat)"
+    )
+    private val externalId by option(
+        ArgType.String,
+        description = "Extra payment identifier specified by the caller"
+    )
+    private val pathFindingExperimentName by option(
+        ArgType.String,
+        description = "Name of the path-finding configuration that should be used"
+    )
+    private val blocking by option(
+        ArgType.Boolean,
+        description = "Block until the payment completes"
+    )
+
+
+    override fun execute() = runBlocking {
+        val eclairClient = eclairClientBuilder.build(host, password)
+
+        val result = eclairClient.payinvoice(
+            invoice!!,
+            amountMsat,
+            maxAttempts,
+            maxFeeFlatSat,
+            maxFeePct,
+            externalId,
+            pathFindingExperimentName,
+            blocking
+        ).map { response ->
+            val result = if (response.length == 36) {
+                PayInvoiceResult(true, response)
+            } else {
+                PayInvoiceResult(false, response)
+            }
+            Serialization.encode(result)
```

Comment:
No, you should not rely on the response length, this is very brittle and will break if one of the field changes its encoding (or if there are optional tlvs that are filled).

You should instead first try to deserialize as a successful response, then if that fails try to deserialize as a failure response, then fallback to a generic error.

Reply:
> No, you should not rely on the response length, this is very brittle and will break if one of the field changes its encoding (or if there are optional tlvs that are filled).
> 
> You should instead first try to deserialize as a successful response, then if that fails try to deserialize as a failure response, then fallback to a generic error.

Just a small doubt, 
Since what the API responds to us is a `uuid` which basically tells us `id` of an attempt to pay the invoice. Now there's a regex for `uuid` which is given on [stackoverflow](https://stackoverflow.com/questions/136505/searching-for-uuids-in-text-with-regex), that is `^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$`

So we can just check if the response that we get is a `uuid` or not.
```
.map { response ->
                val result = if (response.matches(Regex("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"))) {
                    InvoiceResult(true, response)
                } else {
                    InvoiceResult(false, response)
                }
                Serialization.encode(result)
            }
```
The response after implementing this is:
```
claddy@claddy:~/Documents/eclair-cli$ eclair-cli.kexe payinvoice -p password --invoice lnbcrt5n1pjd6c09pp5c7ueze5hk8r2qupkcn7canx8dzv9qplhpye372hu4unjndmn3ftqdqufejhw5rp095kuan0d93k24r9wd6qsp5cxrvzp7l207he3lursph0nx2se9tdgryjyguz8c69ty2z7wute6qmqz9gxqrrsscqp79q7sqqqqqqqqqqqqqqqqqqqsqqqqqysgqwgzsh3nahdym4h0u7q85axffcarvem679x98uk3a43xgy8fwxdm85dy5ty0u46tthclf4082e7ph0jchz0z2ywhp6pf4dw9qrucdrcgpf6eqes --amountMsat 500
{
  "type": "types.InvoiceResult",
  "success": true,
  "message": "5687c787-191b-4184-be07-a32ccf1a015e"
}
```